frontend/src/pages/RunDetailsV2.tsx

```diff
@@ -232,19 +222,9 @@ export function RunDetailsV2(props: RunDetailsV2Props) {
         )}
 
         {/* Pipeline Spec tab */}
-        {selectedTab === 2 && isTemplatePullSuccessFromRun && (
+        {selectedTab === 2 && (
           <div className={commonCss.codeEditor} data-testid={'spec-ir'}>
-            <Editor
-              value={jsyaml.safeDump(jsyaml.safeLoad(templateString || ''))} // Use safeLoad and then safeDump to make sure the Pipeline Spec is in Yaml Form.
-              height={editorHeightWidth}
-              width={editorHeightWidth}
-              mode='yaml'
-              theme='github'
-              editorProps={{ $blockScrolling: true }}
-              readOnly={true}
-              highlightActiveLine={true}
-              showGutter={true}
-            />
+            <PipelineSpecTabContent templateString={templateString!} />
```

Comment:
Should we check if templateString exists before showing this content?
For example: `templateString={templateString || ''}`.

Also apply the same to other pages.

Reply:
Yes, that makes sense!